hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/EthereumTransactionHandler.java

```diff
@@ -64,41 +72,55 @@ public EthereumTransactionHandler(
     }
 
     @Override
-    public void preHandle(@NonNull final PreHandleContext context) {
+    public void preHandle(@NonNull final PreHandleContext context) throws PreCheckException {
         requireNonNull(context);
         final var body = context.body().ethereumTransactionOrThrow();
         final var fileStore = context.createStore(ReadableFileStore.class);
         final var hederaConfig = context.configuration().getConfigData(HederaConfig.class);
-        final var ethTxData = callDataHydration
-                .tryToHydrate(body, fileStore, hederaConfig.firstUserEntity())
-                .ethTxData();
-        if (ethTxData != null) {
-            // Ignore the return value; we just want to cache the signature for use in handle()
-            ethereumSignatures.computeIfAbsent(ethTxData);
-        }
+        final var hydratedTx = callDataHydration.tryToHydrate(body, fileStore, hederaConfig.firstUserEntity());
+
+        validateTruePreCheck(hydratedTx.status() == OK, hydratedTx.status());
+
+        final var ethTxData = hydratedTx.ethTxData();
+        validateTruePreCheck(ethTxData != null, INVALID_ETHEREUM_TRANSACTION);
+
+        // Ignore the return value; we just want to cache the signature for use in handle()
+        ethereumSignatures.computeIfAbsent(ethTxData);
     }
 
     @Override
     public void handle(@NonNull final HandleContext context) throws HandleException {
         // Create the transaction-scoped component
         final var component = provider.get().create(context, ETHEREUM_TRANSACTION);
 
-        // Run its in-scope transaction and get the outcome
-        final var outcome = component.contextTransactionProcessor().call();
+        final var hevmTransactionFactory = component.contextTransactionProcessor().hevmTransactionFactory;
```

Comment:
Hmm...we don't need to do all this to validate the sender nonce. We can just add the validation [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/TransactionProcessor.java#L235) like,
```
        if (transaction.isEthereumTransaction()) {
            validateTrue(transaction.nonce() == parties.sender().getNonce(), WRONG_NONCE);
            parties.sender().incrementNonce();
        }
```
and then add translation for `WRONG_NONCE` as one more `else if` branch [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/hevm/HederaEvmTransactionResult.java#L142),
```
    private static final Bytes WRONG_NONCE_REASON = Bytes.wrap(WRONG_NONCE.name());
    ...
                } else if (revertReason.equals(WRONG_NONCE_REASON)) {
                return WRONG_NONCE;

```
etc.